frontend/public/components/hpa.tsx

```diff
@@ -31,8 +35,98 @@ const menuActions = [
   ...common,
 ];
 
-export const HorizontalPodAutoscalersDetails: React.SFC<HorizontalPodAutoscalersDetailsProps> = ({obj: hpa}) => {
-  return <div className="co-m-pane__body">
+const humanizeResourceValue = (value: string, type: string) => {
+  if (!value) {
+    return null;
+  }
+
+  switch (type) {
+    case 'cpu':
+      return humanizeCPU(value);
+    case 'memory':
+      return humanizeMem(value);
+    default:
+      return value;
+  }
+};
+
+const MetricsTable: React.SFC<MetricsTableProps> = ({obj: hpa}) => {
+  const namespace = hpa.metadata.namespace;
+  // https://github.com/kubernetes/api/blob/master/autoscaling/v2beta1/types.go
+  const metrics = hpa.spec.metrics.map((metric, i) => {
+    let typeLabel, targetValue, currentValue;
+    const currentMetrics = _.get(hpa, ['status', 'currentMetrics', i]);
+    switch (metric.type) {
+      case 'External':
+        typeLabel = metric.external.metricName;
+        // TODO: show metric selector for external metrics?
+        if (metric.external.targetAverageValue) {
+          currentValue = _.get(currentMetrics, 'object.currentAverageValue');
+          targetValue = metric.external.targetAverageValue;
+        } else if (metric.external.targetValue) {
+          currentValue = _.get(currentMetrics, 'object.currentValue');
+          targetValue = metric.external.targetValue;
+        }
+        break;
+      case 'Object':
+        typeLabel = [`${metric.object.metricName} on`, <ResourceLink kind={metric.object.target.kind} name={metric.object.target.name} namespace={namespace} title={metric.object.target.name} key={i} />];
+        targetValue = metric.object.targetValue;
+        currentValue = _.get(currentMetrics, 'object.currentValue');
+        break;
+      case 'Pods':
+        typeLabel = `${metric.pods.metricName} on pods`;
+        currentValue = _.get(currentMetrics, 'pods.currentAverageValue');
+        targetValue = metric.pods.targetAverageValue;
+        break;
+      case 'Resource':
+        typeLabel = `resource ${metric.resource.name}`;
+        if (metric.resource.targetAverageUtilization) {
+          typeLabel += ' (as a percentage of request)';
+          const utilization = _.get(currentMetrics, 'resource.currentAverageUtilization');
+          if (_.isNumber(utilization)) {
+            currentValue = `${utilization}%`;
+          }
+          targetValue = `${metric.resource.targetAverageUtilization}%`;
+        } else if (metric.resource.targetAverageValue) {
+          const currentValueRaw = _.get(currentMetrics, 'resource.currentAverageValue');
+          currentValue = humanizeResourceValue(currentValueRaw, metric.resource.name);
+          targetValue = humanizeResourceValue(metric.resource.targetAverageValue, metric.resource.name);
```

Comment:
It looks like our use of humanize is wrong. Debugging now.

Reply:
`humanizeMem` assumes you're passing the value in bytes instead of a string like `50Mi`. I'd rather not do that since the value in spec is likely using the units we want to show (e.g., we shouldn't convert 50MB to MiB and not have a round number).

I think it might be easiest for the moment to remove the humanize for current value as well since otherwise they will have different formatting (current: 30.22 MiB, target: 50Mi). We can always revisit this.